airbyte-cdk/python/unit_tests/sources/declarative/requesters/test_http_requester.py

```diff
@@ -342,52 +342,84 @@ def test_send_request_params(provider_params, param_params, authenticator_params
             "k=a%2Cb",  # k=a,b
             id="test-request-parameter-comma-separated-strings",
         ),
+        pytest.param(
+            {"k": '{{ config["k"] }}'},
+            {"k": {"updatedDateFrom": "2023-08-20T00:00:00Z", "updatedDateTo": "2023-08-20T23:59:59Z"}},
+            # {'updatedDateFrom': '2023-08-20T00:00:00Z', 'updatedDateTo': '2023-08-20T23:59:59Z'}
+            "k=%7B%27updatedDateFrom%27%3A+%272023-08-20T00%3A00%3A00Z%27%2C+%27updatedDateTo%27%3A+%272023-08-20T23%3A59%3A59Z%27%7D",
+            id="test-request-parameter-from-config-object",
+        ),
+    ],
+)
+def test_request_param_interpolation(request_parameters, config, expected_query_params):
+    options_provider = InterpolatedRequestOptionsProvider(
+        config=config,
+        request_parameters=request_parameters,
+        request_body_data={},
+        request_headers={},
+        parameters={},
+    )
+    requester = create_requester()
+    requester._request_options_provider = options_provider
+    requester.send_request()
+    sent_request: PreparedRequest = requester._session.send.call_args_list[0][0][0]
+    assert sent_request.url.split("?", 1)[-1] == expected_query_params
+
+
+@pytest.mark.parametrize(
+    "request_parameters, config, invalid_value_for_key",
+    [
         pytest.param(
             {"k": "[1,2]"},
             {},
-            "k=1&k=2",
```

Comment:
why are these test cases removed?

Reply:
If I understood correctly, according to the acceptance criterion, `An error is thrown by the low-code CDK if an array or object is attempted to be passed in as a query parameter to the request`, now will not be able to pass this type of data.
Currently, in the master branch, user can pass an array of strings/int/float, and in this case we will have the following examples:
1. `{"k": "[1,2]"} = "k=1&k=2"`
2.  `{"k": ["param1", "param2"]} = "k=param1&k=param2"`
3. `{"k": [12.55, 13.48]} = "k=12.55&k=13.48"`

If we want to keep the ability to pass an array of strings/int/float, but only throw an error if an array of arrays or an array of objects is passed, you need to modify the check conditions a bit.
In this case, we will get an error only if we pass it: 
1. `{"k": [[1,2], [3, 4]]} = ValueError`
2. `{"k": [{3: 4}]} = ValueError`

And objects will throw an error in any case.
`{"k": {"param1": "some_value"}} = ValueError`